src/main/java/uk/gov/companieshouse/overseasentitiesapi/controller/OverseasEntitiesDataController.java

```diff
@@ -106,7 +97,57 @@ private ResponseEntity<OverseasEntityDataApi> getOverseasEntityDataResponse(
             overseasEntityDataApi = new OverseasEntityDataApi();
             overseasEntityDataApi.setEmail(email);
         }
-        
+
         return ResponseEntity.ok(overseasEntityDataApi);
     }
+
+    @GetMapping("/managing-officers")
+    public ResponseEntity<ManagingOfficerListDataApi> getManagingOfficers(
+            @PathVariable(TRANSACTION_ID_KEY) String transactionId,
+            @PathVariable(OVERSEAS_ENTITY_ID_KEY) String overseasEntityId,
+            @RequestHeader(value = ERIC_REQUEST_ID_KEY) String requestId) throws ServiceException {
+
+        HashMap<String, Object> logMap = createLogMap(transactionId, overseasEntityId);
+        ApiLogger.infoContext(requestId, "Calling service to retrieve the overseas entity submission for overseas entity " + overseasEntityId, logMap);
+
+        final var submissionDtoOptional = overseasEntitiesService.getOverseasEntitySubmission(overseasEntityId);
+
+        if (submissionDtoOptional.isPresent()) {
+            final var submissionDto = submissionDtoOptional.get();
+
+            if (!isRoeUpdateEnabled) {
+                throw new ServiceException("ROE Update feature must be enabled for get overseas entity details");
+            }
+
+            return retrieveAndEvaluateManagingOfficerData(submissionDto, overseasEntityId, requestId, logMap);
+        } else {
+            ApiLogger.errorContext(requestId, "Could not find overseas entity submission for overseas entity " + overseasEntityId, null, logMap);
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    private ResponseEntity<ManagingOfficerListDataApi> retrieveAndEvaluateManagingOfficerData(OverseasEntitySubmissionDto submissionDto, String overseasEntityId, String requestId, HashMap<String, Object> logMap) {
+        try {
+            ManagingOfficerListDataApi managingOfficerDataList = privateDataRetrievalService.getManagingOfficerData(submissionDto.getEntityNumber());
+
+            if (managingOfficerDataList == null || managingOfficerDataList.getManagingOfficerData() == null || managingOfficerDataList.getManagingOfficerData().isEmpty()) {
+                ApiLogger.errorContext(requestId, "Could not find any managing officers data for overseas entity " + overseasEntityId, null, logMap);
+                return ResponseEntity.notFound().build();
+            }
+
+            ApiLogger.infoContext(requestId, "Successfully retrieved the managing officers data", logMap);
+            return ResponseEntity.ok(managingOfficerDataList);
+
+        } catch (ServiceException e) {
+            ApiLogger.errorContext(requestId, e.getMessage(), e, logMap);
+            return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
```

Comment:
If we add createLogMap() then use in other PR too. Can do when merging in order.